plugin/src/main/java/org/opensearch/ml/action/controller/DeleteModelControllerTransportAction.java

```diff
@@ -0,0 +1,249 @@
+                String[] targetNodeIds = getAllNodes();
+                MLUndeployModelControllerNodesRequest undeployModelControllerNodesRequest = new MLUndeployModelControllerNodesRequest(
+                    targetNodeIds,
+                    modelId
+                );
+                client
+                    .execute(
+                        MLUndeployModelControllerAction.INSTANCE,
+                        undeployModelControllerNodesRequest,
+                        ActionListener.runBefore(ActionListener.wrap(nodesResponse -> {
+                            if (nodesResponse != null && isUndeployModelControllerSuccessOnAllNodes(nodesResponse)) {
+                                log
+                                    .info(
+                                        "Successfully undeploy model controller from cache. Start to delete the model controller for model {}",
+                                        modelId
+                                    );
+                                deleteModelController(modelId, actionListener);
+                            } else {
+                                String[] nodeIds = getUndeployModelControllerFailedNodesList(nodesResponse);
+                                log
+                                    .error(
+                                        "Failed to undeploy model controller with model ID {} on following nodes {}, deletion is aborted. Please retry or undeploy the model manually and then perform the deletion.",
+                                        modelId,
+                                        Arrays.toString(nodeIds)
+                                    );
+                                actionListener
+                                    .onFailure(
+                                        new RuntimeException(
+                                            "Failed to undeploy model controller with model ID "
+                                                + modelId
+                                                + " on following nodes "
+                                                + Arrays.toString(nodeIds)
+                                                + ", deletion is aborted. Please retry or undeploy the model manually and then perform the deletion."
+                                        )
+                                    );
+                            }
+                        }, e -> {
+                            log
+                                .error(
+                                    "Failed to undeploy model controller from cache and delete the model controller for model {}",
+                                    modelId,
+                                    e
+                                );
+                            actionListener.onFailure(e);
+                        }), context::restore)
+                    );
+            } else {
+                deleteModelController(modelId, actionListener);
```

Comment:
So this method also support delete model for not deployed model ?  If yes, the method name is not accurate, just change to `deleteModelControllerWithDeployedModel`. If you want to highlight that you have some special logic for deployed model, please add more comments. no need to add everything to the method name.

Reply:
Ok I will add comments to it, because `deleteModelController` more fits to the direct delete method I think